sklearn/base.py

```diff
@@ -294,9 +327,33 @@ def set_params(self, **params):
 
     def __repr__(self):
         class_name = self.__class__.__name__
-        return '%s(%s)' % (class_name, _pprint(self.get_params(deep=False),
+        params = self.get_params(deep=False)
+        # Consider the constructor parameters excluding 'self'
+        if _PRINTOPTIONS['parameters'] == 'changed':
+            filtered_params = {}
+            init_params = signature(self.__init__).parameters
+            for k, v in params.items():
+                if v != init_params[k].default:
+                    filtered_params[k] = v
+            params = filtered_params
+        return '%s(%s)' % (class_name, _pprint(params,
                                                offset=len(class_name),),)
 
+    def _repr_pretty_(self, p, cycle):
```

Comment:
How would we do that? I removed ``__str__`` to get as little "side effects" as possible. We could a (private?) method ``_info``?